### PR TITLE
Update manual testing section in DG

### PR DESCRIPTION
**Pull request by minosx31**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -32,21 +33,36 @@ title: Developer Guide
...
+
+:information_source: **Notes:** Notes aim to provide you with extra information.
+
+:bulb: **Tip:** Tips are useful suggestions that you can follow.
 
 ## **Acknowledgements**
 
-- {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- This project is based on the AddressBook-Level3 project created by the [SE-EDU initiative](https://se-education.org).
+- Libraries used: [JavaFX](https://openjfx.io/), [Jackson](https://github.com/FasterXML/jackson), [JUnit5](https://github.com/junit-team/junit5)
```
Comment (minosx31) [inline docs/DeveloperGuide.md:56]: @niekis you might wanna add Opencsv here. Can also provide me the source link so that I can add in for you

Comment (rexcyrio) [reply docs/DeveloperGuide.md:56]: https://opencsv.sourceforge.net

Source: https://github.com/nus-cs2103-AY2223S2/forum/issues/271

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -568,22 +612,118 @@ testers are expected to do more *exploratory* testing.
...
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
-1. *{ more test cases …​ }*
+
+[Return to Top](#table-of-contents)
+
+### Toggling the favorite status of a person
+
+1. Toggling the favorite status of a person while all persons are being shown
+   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
```
Comment (rexcyrio) [inline docs/DeveloperGuide.md:705]: I think another prerequisite should be that there are no currently favorited persons in the favorites list.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -568,22 +612,118 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: No person is toggled as favorite. Error details shown in the message box.
+
+[Return to Top](#table-of-contents)
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
-   1. *{explain how to simulate a missing/corrupted file, and the expected behavior}*
-1. *{ more test cases …​ }*
+   1. Prerequisite: Delete the `aims.json` file in `./data` folder and launch AIMS again.<br>
```
Comment (rexcyrio) [inline docs/DeveloperGuide.md:718]: > Delete the `aims.json` file in `./data` folder and launch AIMS again.

This sounds like the test case itself

I think for the prerequisite we can say that "the `aims.json` file exists in the `./data` folder"

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -568,22 +612,118 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: AIMS loads a default list of persons and the data file is created again.
```
Comment (rexcyrio) [inline docs/DeveloperGuide.md:719]: Do note that the new `aims.json` data file is only created **after** the user exits the app (i.e., when the user is still using the app, the newly created `aims.json` does not exist on disk yet)

**Review by rexcyrio — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -568,22 +612,118 @@ testers are expected to do more *exploratory* testing.
...
+   ``` text
+   add n/Alex Yeoh
+   r/CPL
+   c/Bravo
+   u/1 SIR
+   p/89043761
+   e/alexyeoh@example.com
+   a/Blk 21 Buona Vista Rd, #11-01
+   ```
+   Expected: Error details shown in the message box, indicating that duplicate emails or phones are not allowed.
```
[the remark this answers is not in the document]
Comment (minosx31) [reply docs/DeveloperGuide.md]: I was following their format hehe but ig i can add a note for this?

**Review by rexcyrio — APPROVED** [review]
LGTM :+1:

**Pull request merged**